test/python/quantum_info/states/test_stabilizerstate.py

```diff
@@ -740,14 +740,30 @@ def test_expval_random_subsystem(self, num_qubits):
 
         for _ in range(self.samples):
             cliff = random_clifford(num_qubits, seed=self.rng)
-            op = random_pauli(2, seed=self.rng)
+            op = random_pauli(2, group_phase=True, seed=self.rng)
             qargs = np.random.choice(num_qubits, size=2, replace=False)
             qc = cliff.to_circuit()
             stab = StabilizerState(cliff)
             exp_val = stab.expectation_value(op, qargs)
             target = Statevector(qc).expectation_value(op, qargs)
             self.assertAlmostEqual(exp_val, target)
 
+    @combine(num_qubits=[2, 3, 4, 5])
+    def test_expval_from_random_clifford(self, num_qubits):
+        """Test that the expectation values for a random Clifford,
+        where the Pauli operators are all its stabilizers,
+        are equal to 1."""
+
+        for _ in range(self.samples):
+            cliff = random_clifford(num_qubits, seed=self.rng)
+            qc = cliff.to_circuit()
+            stab = StabilizerState(qc)
+            stab_gen = stab.clifford.to_dict()["stabilizer"]
+            for i in range(num_qubits):
+                op = Pauli(stab_gen[i])
+                exp_val = stab.expectation_value(op)
+                self.assertEqual(exp_val, 1)
+
```

Comment:
Perhaps we could extend this test to run through cases where the `Pauli` is given each of the four possible phases as well, and test those?

Reply:
This test is specific to Cliffords, where the operator to evaluate the expectation value is one of its stabilizers, and then the expectation value should always be 1. This is how this bug was detected, so I added it as a test.

In addition, I changed the tests `test_expval_random` and `test_expval_random_subsystem`, where lines 730 and 743 were replaced to:
`op = random_pauli(num_qubits, group_phase=True, seed=self.rng)`
so that the operator can be any random Pauli (including, a minus or i/-i sign) and hence the expectation value can also be -1,i or -i.

Reply:
In addition, I added some deterministic checks in `test_expval_single_qubit` and `test_expval_two_qubits` to include expectation values of -1,i and -i.